src/backend/distributed/commands/text_search.c

```diff
@@ -132,6 +141,14 @@ PostprocessCreateTextSearchDictionaryStmt(Node *node, const char *queryString)
 	EnsureSequentialMode(OBJECT_TSDICTIONARY);
 
 	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
+
+	DeferredErrorMessage *errMsg = DeferErrorIfHasUnsupportedDependency(&address);
+	if (errMsg != NULL)
+	{
+		RaiseDeferredError(errMsg, WARNING);
+		return NIL;
+	}
+
 	EnsureDependenciesExistOnAllNodes(&address);
 
 	QualifyTreeNode(node);
```

Comment:
It is not related to this PR but @hanefi can you please check whether we need to move `QualifyTreeNode` to the preprocess function with a separate PR? Check the discussion from https://github.com/citusdata/citus/pull/5827#discussion_r830030018

Reply:
I do not think that a change is necessary here. We do not have a preprocess function here. We only create the Dictionary objects in the workers after they are successfully created on the coordinator node. Since we do not send any queries to workers in the (non-existent) preprocess step, I do not think that we need to qualify the parsed `Node *node`.

Reply:
QualifyTreeNode must always be in preprocess, because its main purpose it to ensure that we use the same schema in Citus code and the internal ProcessUtility function, even if search_path changes in between.

Reply:
Does this mean that we should introduce preprocess steps for all `CREATE ...` commands?

We do not have those functions for at least text search dictionaries and configurations. Should we introduce them?

It may worth checking all `PreprocessDefine.*Stmt` functions. I also see that some of them are missing `QualifyTreeNode` calls, such as `PreprocessDefineCollationStmt`

Reply:
might be related: https://github.com/citusdata/citus/issues/5839